### PR TITLE
MDEXP-628 - data-export/export: Holdings Strategy

### PR DESCRIPTION
**Pull request by alekGbuz**
## Purpose

[MDEXP-628](https://issues.folio.org/browse/MDEXP-628) - data-export/export: Holdings Strategy
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] Check logging
  - [ ] There are no major code smells or security issues

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dataexp/domain/entity/JobExecutionExportFilesEntity.java [hunk 1]
```diff
@@ -35,4 +36,10 @@ public class JobExecutionExportFilesEntity {
 
   @Enumerated(EnumType.STRING)
   private JobExecutionExportFilesStatus status;
+
+  public void setStatusBaseExportStatistic(ExportStrategyStatistic exportStatistic) {
+    if (exportStatistic.getFailed() == 0 && exportStatistic.getExported() > 0) this.status = JobExecutionExportFilesStatus.COMPLETED;
```
Comment (khandramai) [inline src/main/java/org/folio/dataexp/domain/entity/JobExecutionExportFilesEntity.java:41]: Please, follow standard convention for if-statements https://www.oracle.com/java/technologies/javase/codeconventions-statements.html.

File: src/main/java/org/folio/dataexp/config/ExportConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  @Bean
+  public RuleProcessor ruleProcessor() {
+    return new RuleProcessor(TranslationsFunctionHolder.SET_VALUE);
+  }
+
+  @Bean
+  public List<Rule> defaultRulesFromConfigFile() throws IOException {
+    var mapper = new ObjectMapper();
+    try (InputStream is = ExportConfiguration.class.getResourceAsStream("/rules/rulesDefault.json")) {
```
Comment (khandramai) [inline src/main/java/org/folio/dataexp/config/ExportConfiguration.java:28]: "/rules/rulesDefault.json" - "magic path". Can this be a constant?

File: src/main/java/org/folio/dataexp/config/ExportConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    } catch (IOException e) {
+      log.error("Failed to fetch default rules for export");
+      throw e;
+    }
+  }
+
+  @Bean
+  public List<Rule> defaultHoldingsRulesFromConfigFile() throws IOException {
+    var mapper = new ObjectMapper();
+    try (InputStream is = ExportConfiguration.class.getResourceAsStream("/rules/holdingsRulesDefault.json")) {
```
Comment (khandramai) [inline src/main/java/org/folio/dataexp/config/ExportConfiguration.java:40]: "/rules/holdingsRulesDefault.json" - "magic path". Can this be a constant?

**Review by khandramai — CHANGES_REQUESTED** [review]
Please, follow Java code convention for statements https://www.oracle.com/java/technologies/javase/codeconventions-statements.html

File: src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportStrategy.java [hunk 4]
```diff
@@ -4,26 +4,24 @@
...
   private final JobExecutionExportFilesEntityRepository jobExecutionExportFilesEntityRepository;
+  private final FolioS3Client s3Client;
 
   @Override
   public ExportStrategyStatistic saveMarcToRemoteStorage(JobExecutionExportFilesEntity exportFilesEntity) {
     var marc = "marc";
-    var s3Client = folioS3ClientFactory.getFolioS3Client();
     var remoteStorageWriter = new RemoteStorageWriter(exportFilesEntity.getFileLocation(),  8192, s3Client);
     remoteStorageWriter.write(marc);
     remoteStorageWriter.close();
```
Comment (obozhko-folio) [inline src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportStrategy.java:23]: This can be done with try-with-resources.

**Review by obozhko-folio — APPROVED** [review]

File: src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportStrategy.java (hunk 4 above)
Comment (alekGbuz) [reply src/main/java/org/folio/dataexp/service/export/strategies/InstancesExportStrategy.java]: Updated thanks

File: src/main/java/org/folio/dataexp/domain/entity/JobExecutionExportFilesEntity.java (hunk 1 above)
Comment (alekGbuz) [reply src/main/java/org/folio/dataexp/domain/entity/JobExecutionExportFilesEntity.java]: Updated thanks.

**Review by khandramai — APPROVED** [review]

**Pull request merged**